### PR TITLE
eapi8-dosym.eclass

### PR DESCRIPTION
**Pull request by ulm**

### CODE REVIEW

File: eclass/eapi8-dosym.eclass [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		[[ ${target} == /* ]] \
+			|| die "${FUNCNAME}: -r specified but no absolute target path"
+
+		target=$(_dosym8_canonicalize "${target}")
+		dir=$(_dosym8_canonicalize "/${link#/}")
+		dir=${dir%/*}			# poor man's dirname
+		dir=${dir:-/}			# use "/" if we ended up with a null string
+
+		while [[ -n ${dir} ]]; do
+			c=${dir%%/*}
```
Comment (mgorny) [inline eclass/eapi8-dosym.eclass:96]: This is the part where more verbose variable names would be helpful.

File: eclass/tests/eapi8-dosym.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+source tests-common.sh
+
+inherit eapi8-dosym
+
+dosym() {
+    echo "$1"
```
Comment (mgorny) [inline eclass/tests/eapi8-dosym.sh:12]: Space indent.

Comment (ulm) [reply eclass/tests/eapi8-dosym.sh]: Fixed.

File: eclass/eapi8-dosym.eclass (hunk 1 above)
Comment (ulm) [reply eclass/eapi8-dosym.eclass]: I went for `linkdir` and `comp`.

File: eclass/tests/eapi8-dosym.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	teq b/f ${f} /a/b///./c/d/../e/..//../f /a/././///g/../h
+	teq ../h ${f} /a/././///g/../h /a/b///./c/d/../e/..//../f
+done
+
+randompath() {
+	dd if=/dev/urandom count=1 2>/dev/null \
+		| LC_ALL=C sed 's/[^a-zA-N]//g;s/[A-E]/\/.\//g;s/[F-J]/\/..\//g;s/[K-N]/\//g;s/^/\//;q'
+}
+
+# set the "crazy" variable to a positive number to enable crazy random tests
```
[the remark this answers is not in the document]
Comment (ulm) [reply eclass/tests/eapi8-dosym.sh:63]: Yeah, don't take this too serious. :smile:
Though it exposed a bug for some weird corner case.

File: eclass/tests/eapi8-dosym.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
```
Comment (SoapGentoo) [inline eclass/tests/eapi8-dosym.sh:1]: portable shebang pls ;)

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: eclass/tests/eapi8-dosym.sh (hunk 4 above)
Comment (ulm) [reply eclass/tests/eapi8-dosym.sh:1]: It's the same as in all other tests. And I hate these `env` shebangs (which assumes that `env` is installed in a particular location, so just shifts the problem).

Comment (SoapGentoo) [reply eclass/tests/eapi8-dosym.sh:1]: what if I run these tests on some old CentOS 6 Prefix installation?

Comment (mgorny) [reply eclass/tests/eapi8-dosym.sh:1]: Or on NetBSD where bash is in /usr/pkg/bin?

Comment (ulm) [reply eclass/tests/eapi8-dosym.sh:1]: Yeah, sure. Continue like this and I'm going to drop tests altogether. :smile:

It's `/bin/bash` everywhere, and `tests-common.sh` sources `/lib/gentoo/functions.sh`. Does the latter exist on Prefix?

Comment (SoapGentoo) [reply eclass/tests/eapi8-dosym.sh:1]: we should make that relative then

Comment (ulm) [reply eclass/tests/eapi8-dosym.sh:1]: This is all part of a global issue that has little relation to this particular eclass and its tests.
